src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -570,7 +570,7 @@ def get_versioned_sdk_path(api_profile, resource_type, operation_group=None):
 
 
 def get_versioned_sdk(api_profile, resource_type, *attr_args, **kwargs):
-    checked = kwargs.get('checked', True)
+    checked = kwargs.get('checked', None)
```

Comment:
The current implementation is indeed a design flaw that can't distinguish

- expected import failure caused by incompatible SDK profile (API version)
- corrupted SDK installation

When the SDK installation is corrupted, this will trigger unclear error message like https://github.com/Azure/azure-cli/issues/14054 in many places:

https://github.com/Azure/azure-cli/blob/045bbfbaa3ecbd1c86d109601f99bc1e90af029d/src/azure-cli/azure/cli/command_modules/vm/_params.py#L117

https://github.com/Azure/azure-cli/blob/0804c756640bb652556d3917961bb39003197a63/src/azure-cli/azure/cli/command_modules/vm/custom.py#L301-L302

A better pattern is to

1. call `get_models` lazily only when `min_api` or `max_api` is satisfied
2. error out if import failure happens

This requires large amount of effort inspecting the whole code base.